### ISSUE TITLE
Weird issue when trying to build

### ISSUE THREAD

**Issue by ChemicalRocketeer** [pkg/sftp#281]
If I run `go test`, it fails with exit code 2, but none of my tests seem to be failing. The only output I get that looks at all suspicious is this series of errors from the sftp library:

```
# github.com/pkg/sftp
../../../../pkg/mod/github.com/pkg/sftp@v1.8.3/attrs.go:98:14: undefined: unmarshalUint32
../../../../pkg/mod/github.com/pkg/sftp@v1.8.3/attrs.go:105:16: undefined: unmarshalUint64
../../../../pkg/mod/github.com/pkg/sftp@v1.8.3/attrs.go:108:15: undefined: unmarshalUint32
../../../../pkg/mod/github.com/pkg/sftp@v1.8.3/attrs.go:111:15: undefined: unmarshalUint32
../../../../pkg/mod/github.com/pkg/sftp@v1.8.3/attrs.go:114:16: undefined: unmarshalUint32
../../../../pkg/mod/github.com/pkg/sftp@v1.8.3/attrs.go:117:17: undefined: unmarshalUint32
../../../../pkg/mod/github.com/pkg/sftp@v1.8.3/attrs.go:118:17: undefined: unmarshalUint32
../../../../pkg/mod/github.com/pkg/sftp@v1.8.3/client.go:519:41: undefined: StatVFS
../../../../pkg/mod/github.com/pkg/sftp@v1.8.3/client.go:1249:34: undefined: StatusError
../../../../pkg/mod/github.com/pkg/sftp@v1.8.3/conn.go:131:34: undefined: ider
../../../../pkg/mod/github.com/pkg/sftp@v1.8.3/attrs.go:118:17: too many errors
``` 

Does anyone know how I can deal with this? Sorry if it's a stupid question or unrelated, I'm still a bit of a Go newb and I couldn't find anything with google.

**Comment by davecheney** [comment]
Thank you for raising this issue. The first step to diagnosing the problem is to provide a complete reproduction case. It looks like you’re building a program using Go modules that imports this package. 

Can you please list exactly the steps you took that lead to this problem.

**Comment by ChemicalRocketeer** [comment]
Thanks for the response! I wish I could, but this is a previously-existing project and I have no idea how it got this way.

**Comment by davecheney** [comment]
Can you post as much of your terminal scrollback as possible? At a minimum
we need to see the commands you used to fetch the source of the program and
the command you used to test it.

**Comment by ChemicalRocketeer** [comment]
We managed to get it working! Using `GOFLAGS=-mod=vendor`, `rm -rf $GOPATH/pkg/mod`, and `go mod vendor`. I guess my pkg/mod must have had something bad going on.

Other people working in the same codebase didn't have this issue, and it even persisted when I cloned another repo that had the same dependency.

For posterity's sake, here's all I was doing to run the project:

```
git clone <repo>
go mod tidy
go mod vendor
go test ./... -cover -failfast -p 6 #fails
```
